File: phong_tap_fe/quan-ly-phong-gym/src/common/shared/enums/giamGia.enums.ts
export enum GiamGiaEnums {
  NAMPHANTRAM = 1,
  MUOIPHANTRAM = 2,
  MUOILAMPHANTRAM = 3,
  HAIMUOIPHANTRAM = 4,
  HAILAMPHANTRAM = 5,
  NAMMUOIPHANTRAM = 6,
}

export const giamGia: { value: number; label: string }[] = [
  { value: GiamGiaEnums.NAMPHANTRAM, label: '5%' },
  { value: GiamGiaEnums.MUOIPHANTRAM, label: '10%' },
  { value: GiamGiaEnums.MUOILAMPHANTRAM, label: '15%' },
  { value: GiamGiaEnums.HAIMUOIPHANTRAM, label: '20%' },
  { value: GiamGiaEnums.HAILAMPHANTRAM, label: '25%' },
  { value: GiamGiaEnums.NAMMUOIPHANTRAM, label: '50%' },
];

File: phong-tap-be/phong_tap-be/QuanLyPhongGym/src/main/java/com/example/QuanLyPhongGym/domain/enums/GiamGiaEnums.java
package com.example.QuanLyPhongGym.domain.enums;

import java.util.Map;

public enum GiamGiaEnums {
    NAMPHANTRAM(1, "5%"),
    MUOIPHANTRAM(2, "10%"),
    MUOILAMPHANTRAM(3, "15%"),
    HAIMUOIPHANTRAM(4, "20%"),
    HAILAMPHANTRAM(4, "25%"),
    NAMMUOIPHANTRAM(4, "50%");

    public static final Map<Integer, GiamGiaEnums> dict = Map.of(
            GiamGiaEnums.NAMPHANTRAM.value, GiamGiaEnums.NAMPHANTRAM,
            GiamGiaEnums.MUOIPHANTRAM.value, GiamGiaEnums.MUOIPHANTRAM,
            GiamGiaEnums.MUOILAMPHANTRAM.value, GiamGiaEnums.MUOILAMPHANTRAM,
            GiamGiaEnums.HAIMUOIPHANTRAM.value, GiamGiaEnums.HAIMUOIPHANTRAM,
            GiamGiaEnums.HAILAMPHANTRAM.value, GiamGiaEnums.HAILAMPHANTRAM,
            GiamGiaEnums.NAMMUOIPHANTRAM.value, GiamGiaEnums.NAMMUOIPHANTRAM);

    public final Integer value;
    public final String label;

    GiamGiaEnums(Integer value, String label) {
        this.value = value;
        this.label = label;
    }
}
